src/util/container.h

```diff
@@ -140,6 +140,13 @@ class MutexedQueue
 		m_signal.post();
 	}
 
+	void emplace_back(T &&t)
+	{
+		MutexAutoLock lock(m_mutex);
+		m_queue.emplace_back(std::move(t));
+		m_signal.post();
+	}
```

Comment:
You don't need to emplace to move-insert. See eg. https://en.cppreference.com/w/cpp/container/deque/push_back .
```suggestion
	void push_back(T &&t)
	{
		MutexAutoLock lock(m_mutex);
		m_queue.push_back(std::move(t));
		m_signal.post();
	}
```

Reply:
Good point and I should have known this, will rename.